SignalServiceKit/src/Contacts/Contact.m

```diff
@@ -345,16 +345,28 @@ - (CNContact *)buildCNContactMergedWithNewContact:(CNContact *)newCNContact
         mergedCNContact.familyName = newCNContact.familyName.ows_stripped;
         mergedCNContact.nameSuffix = newCNContact.nameSuffix.ows_stripped;
     }
-    
+
+    if (mergedCNContact.organizationName.ows_stripped.length < 1) {
+        mergedCNContact.organizationName = newCNContact.organizationName.ows_stripped;
+    }
+
     // Phone Numbers
-    NSSet<PhoneNumber *> *existingPhoneNumberSet = [NSSet setWithArray:self.parsedPhoneNumbers];
-    
+    NSMutableSet<PhoneNumber *> *existingPhoneNumberSet = [NSMutableSet setWithArray:self.parsedPhoneNumbers];
+    [existingPhoneNumberSet addObjectsFromArray:self.userTextPhoneNumbers];
```

Comment:
Don't drop unparse-able phone numbers unless they match an existing unparsed phone number.

Phone parsing is hard and I suggest we should be permissive about preserving phone numbers we don't understand. WDYT?

Reply:
That seems fine. 

PhoneNumberSet *was* a set of PhoneNumbers, now it looks like it's a set of PhoneNumber's and NSString's, can you remove the incorrect annotation? Or make it an NSObject.

Reply:
Whoops, fixed.